pyrasterframes/src/main/python/docs/aggregation.pymd

```diff
@@ -10,13 +10,65 @@ import os
+""").select(F.col('id'), rf_tile_mean(F.col('t'))).show(10, False)
+```
+
+In this next code block we are using @ref:[`rf_agg_mean`](reference.md#rf-agg-mean) to compute the DataFrame aggregate, which averages 25 values of 1.0 and 25 values of 3.0, across the fifty cells in two rows. Note that only a single row is returned since the average is computed over the full DataFrame.
+
+```python
+import pyspark.sql.functions as F
+
+spark.sql("""
+SELECT 1 as id, rf_make_ones_tile(5, 5, 'float32') as t
+UNION
+SELECT 2 as id, rf_local_multiply(rf_make_ones_tile(5, 5, 'float32'), 3.0) as t
+""").agg(rf_agg_mean(F.col('t'))).show(10, False)
+```
+
+
+In this code block, we are using @ref:[`rf_agg_local_mean`](reference.md#rf-agg-local-mean) to compute the element-wise local aggregate mean across the two rows. In this example it is computing the mean of one value of 1.0 and one value of 3.0 to arrive at the element-wise mean, but doing so twenty-five times, one for each position in the `tile`.
 
 ```python
-rf = spark.read.geotiff(os.path.join(resource_dir(), 'L8-B8-Robinson-IL.tiff'))
-rf.show(5, False)
+import pyspark.sql.functions as F
+
+alm = spark.sql("""
+SELECT 1 as id, rf_make_ones_tile(5, 5, 'float32') as t
+UNION
+SELECT 2 as id, rf_local_multiply(rf_make_ones_tile(5, 5, 'float32'), 3) as t
+""").agg(rf_agg_local_mean(F.col('t')).alias('l'))
+alm.select(rf_explode_tiles(alm.l)).show(10, False)
+```
+
+## Cell Counts
+
+We can also count the total number of data and NoData cells over all the tiles in a DataFrame using @ref:[`rf_agg_data_cells`](reference.md#rf-agg-data-cells) and @ref:[`rf_agg_no_data_cells`](reference.md#rf-agg-no-data-cells). There are 3,348,900 data cells and 0 NoData cells in this DataFrame.
 
-stats = rf.agg(rf_agg_no_data_cells('tile'), rf_agg_data_cells('tile'), rf_agg_mean('tile'))
+```python
+rf = spark.read.raster('https://s22s-test-geotiffs.s3.amazonaws.com/luray_snp/B02.tif')
+stats = rf.select(rf_tile("proj_raster").alias("tile")).agg(rf_agg_data_cells('tile'), rf_agg_no_data_cells('tile'))
 stats.show(5, False)
 ```
 
+## Statistical Summaries
+
+The @ref:[`rf_agg_stats`](reference.md#rf-agg-stats) function aggregates over all of the tiles in a DataFrame and returns a statistical summary of cell values: number of data cells, number of NoData cells, minimum, maximum, mean, and variance. You can use the @ref:[`rf_tile_stats`](reference.md#rf-tile-stats) to compute summary statistics separately for each row in a `tile` column and you can use the @ref:[`rf_agg_local_stats`](reference.md#rf-agg-local-stats) to compute the local element-wise summary statistics.
+
+```python
+rf = spark.read.raster('https://s22s-test-geotiffs.s3.amazonaws.com/luray_snp/B02.tif')
+stats = rf.select(rf_tile("proj_raster").alias("tile")).agg(rf_agg_stats('tile').alias("stats"))
+stats.printSchema()
+stats.select("stats.min", "stats.max", "stats.mean", "stats.variance").show(1, False)
+```
```

Comment:
It would be worth a bit more exposition and demonstration about `rf_tile_stats` and `rf_agg_local_stats`; perhaps we can try to make clear the alignment with the different kinds of aggregates we defined above.

Another point, would be nice to have  an additional section,  demonstrating the histogram and displaying a matplotlib plot. Not necessary for MVP but very nice if we could. (Also nice to try and if it's not easy /possible,  put up issues to make such a thing possible / easy.)

Reply:
I'm trying to demonstrate `rf_agg_local_stats`, but I'm getting this error... `Cannot combine rasters with different dimensions.(256,256) does not match (38,256)`. Is that a bug or am I doing something wrong? This is the function call: `rf.agg(rf_agg_local_stats('proj_raster').alias('stats')).select("stats.min", "stats.max", "stats.mean", "stats.variance").show(10, False)`

Reply:
ahh see above comment. you indeed cannot combine tiles (array) of different dimensions (shapes) in aggregation across rows.